_posts/2023-07-11-missing-side-effects-on-selector-access-bug.md

```diff
@@ -0,0 +1,53 @@
+---
+layout: post
+published: true
+title: 'Bug when accessing the .selector member on complex expressions in legacy code generation'
+date: '2023-07-11'
+author: Solidity Team
+category: Security Alerts
+---
+
+On June 26, 2023, a bug in the Solidity legacy code generator was found after further
+investigation of a security report related to the use of ``abi.decode`` with a ternary
+expression as the type argument.
+
+The legacy code generation was not evaluating expressions when accessing the ``.selector``
+member on some complex expressions, like assignments and conditionals, while the ``via-ir`` code
+generation behave as expected and evaluates such expressions.
+This leads to a wrong compiler behavior in the legacy code generation.
+
+The bug was already present since the compiler version ``0.6.2``.
+Despite this, as far as we know, it was never exploited or reported externally, and
+therefore we assigned it a severity of "low".
+
+## Which Contracts are Affected?
+
+If you still using the legacy code generation pipeline and your contract uses the 
+``.selector`` member access on some complex expressions,
+like the return of a function call, such expressions will not be evaluated in the 
+legacy code generation, leading to a different behavior when compiling with and without ``via-ir``.
```

Comment:
```suggestion
If you are still using the legacy code generation pipeline and your contract uses the 
``.selector`` member access on some complex expressions,
like the return value of a function call, such expressions will not be evaluated in the 
legacy code generation, leading to a different behavior when compiling with and without ``via-ir``.
```

Also, this mostly repeats what was already said in the intro. I think you should instead list the specific conditions needed to trigger it.